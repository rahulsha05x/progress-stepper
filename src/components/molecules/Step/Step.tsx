import React from 'react';
import Button from '../../atoms/Button/Button';

interface Props {
  text?: string;
  status?: string;
}

const Step: React.FC<Props> = ({ text, status, children }) => {
  text = text ? text : 'not described';
  status = status ? status : 'Not started';
  let stepClass = ['Step'];
  let stepIconClass = ['Step__Status'];
  stepIconClass = [`Step__Status--${status}`, ...stepIconClass];
  stepClass = [`Step--${status}`, ...stepClass];
  return (
    <div className={stepClass.join(' ')}>
      <i className={stepIconClass.join(' ')} />
      <p className="Step__Description">{text}</p>
      {status === 'Current' && (
        <Button className="Step__Button">Begin Task</Button>
      )}
    </div>
  );
};

export default Step;
