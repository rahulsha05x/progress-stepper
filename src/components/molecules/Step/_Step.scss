.Step {
  display: flex;
  align-items: center;
  position: relative;
  padding: 0 25px;
  border-bottom: 1px solid $light-gray;

  &:last-child {
    &::after {
      display: none;
    }
  }
  &::after {
    content: '';
    position: absolute;
    background-color: $light-gray;
    height: 39px;
    width: 2px;
    left: 33px;
    top: 35px;
    z-index: 2;
  }
  &--Complete {
    &::after {
      background-color: $blue-1;
    }
  }
  &--current {
    &::after {
      background-color: $light-gray;
    }
  }
  &__Status {
    position: relative;
    margin-right: 45px;
    border: 1px solid steelblue;
    height: 16px;
    width: 16px;
    border-radius: 50%;
    z-index: 3;
    &--Pending {
      border: 1px solid $light-gray;
    }
    &--Current {
      border: 1px solid $blue-1;
    }
    &--Complete {
      border: 1px solid $blue-1;
      background-color: $blue-1;
    }
    &:last-child {
      &::after {
        display: none;
      }
      &::before {
        display: none;
      }
    }
    // &::before {
    //   content: '';
    //   position: absolute;
    //   background-color: $blue-1;
    //   height: 17px;
    //   width: 2px;
    //   left: 8px;
    //   bottom: 16px;
    // }
    // &::after {
    //   content: '';
    //   position: absolute;
    //   background-color: $blue-1;
    //   height: 35px;
    //   width: 2px;
    //   left: 8px;
    //   top: 16px;
    // }
  }
}
