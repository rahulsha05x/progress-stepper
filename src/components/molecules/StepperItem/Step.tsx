import React from 'react';
interface StepProps {
  className?: string;
  color?: string;
  completed?: boolean;
  content?: any;
  text?: string;
  children?: any;
}
export const Step = ({ className = '', children }: StepProps) => {
  const getChildren = () => {
    return React.Children.map(children, (child, index) => {
      if (child) {
        return React.cloneElement(child, {
          itemindex: index,
        });
      }
    });
  };
  const child = getChildren();
  return (
    <li data-testid="step" className={`progress-step ${className}`}>
      {child}
    </li>
  );
};
export default Step;
