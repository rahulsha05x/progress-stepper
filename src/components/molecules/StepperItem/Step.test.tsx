import React from 'react';
import { render, screen } from '@testing-library/react';
import Step from './Step';

describe('Step Component', () => {
  test('Renders Step component', () => {
    render(<Step />);
    const stepElement = screen.getByTestId('step');
    expect(stepElement.nodeName).toBe('LI');
    expect(stepElement).toMatchSnapshot();
    expect(stepElement).toHaveClass('progress-step');
  });
  test('Renders Step component with custom class', () => {
    render(<Step className="customClass" />);
    const stepElement = screen.getByTestId('step');
    expect(stepElement).toHaveClass('progress-step', 'customClass');
  });
  test('Renders Step component with child elements', () => {
    const childElement = <div>Child one</div>;
    const childElement2 = <div>Child two</div>;
    render(
      <Step
        className="customClass"
        color="yellow"
        completed
        content="Test step"
      >
        {childElement}
        {childElement2}
      </Step>
    );
    const stepElement = screen.getByTestId('step');
    expect(stepElement).toHaveClass('progress-step', 'customClass');
  });
});
