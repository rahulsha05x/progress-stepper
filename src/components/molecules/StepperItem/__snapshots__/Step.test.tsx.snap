// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Component Renders Step component 1`] = `
<li
  class="progress-step "
  data-testid="step"
/>
`;
